gcp_variant_transforms/bq_to_vcf.py

```diff
@@ -98,21 +99,29 @@ def run(argv=None):
     known_args.number_of_bases_per_shard = sys.maxsize
 
   temp_folder = google_cloud_options.temp_location or tempfile.mkdtemp()
-  timestamp_str = datetime.now().strftime('%Y%m%d_%H%M%S')
+  # TODO(allieychen): In both here and vcf_to_bq, the job name and time are
+  # added to the names of the temp files to ensure uniqueness such that multiple
+  # pipelines can run at the same time. Refactor it to a common lib.
+  job_name = '-'.join([_BQ_TO_VCF_SHARDS_JOB_NAME,
+                       datetime.now().strftime('%Y%m%d-%H%M%S')])
+  if google_cloud_options.job_name:
+    google_cloud_options.job_name += '-' + job_name
```

Comment:
Thanks Asha, I have found some issues in my previous PR and fixed it (the temp files should use `google_cloud_options.job_name` rather than `job_name`). 
Once we agree on using the `google_cloud_options.job_name` as the unique identifier, I think changing the job name is necessary. If the user provides the job name, it is possible that there are already those temp files (either not deleted or in the composing stage), so adding the time stamp would ensure unique files. If the user didn't provide the job_name, well, then we only have time stamp as the unique identifier. By raising errors that two job names cannot be the same it is ensured that the temp files are unique. It is truly complex though.

Reply:
As discussed offline, please add a uuid to the end; we should do the same for vcf headers too (once we merge them together).
```
unique_temp_id =  '-'.join([google_cloud_options.job_name or _BQ_TO_VCF_SHARDS_JOB_NAME,
datetime.now().strftime('%Y%m%d-%H%M%S'),
str(uuid.uuid4())])
```